@here/harp-mapview/lib/text/TextElementsRenderer.ts

```diff
@@ -1646,17 +1635,41 @@ export class TextElementsRenderer {
             const textIsOptional: boolean =
                 pointLabel.poiInfo !== undefined && pointLabel.poiInfo.textIsOptional === true;
 
-            const textIsFadingIn = textRenderState!.isFadingIn();
-            const textIsFadingOut = textRenderState!.isFadingOut();
-            const textSpaceAvailable = !this.m_screenCollisions.isAllocated(tempBox2D);
-            const textVisible =
-                groupState.visited &&
-                (pointLabel.textMayOverlap ||
-                    textSpaceAvailable ||
-                    textIsFadingIn ||
-                    textIsFadingOut);
-
-            if (textVisible) {
+            const textSpaceAvailable =
```

Comment:
at some stage, it would be good to split this into two sections, to manage the fade in / visibility state for the icons and then separately the text

Reply:
I'm progressing into something like that in my next PRs. Nevertheless, the state of icon and text are not completely independent: if the icon is rejected the text is rejected as well, and if text is rejected the icon is rejected if the text is not optional.